src/components/LinePlotContainer/LinePlotContainer.tsx

```diff
@@ -3,66 +3,51 @@ import { Grid, CardContent, Card, ButtonGroup, Button, Paper } from '@material-u
 
 import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos'
 import LinePlot from '@components/LinePlot/LinePlot'
-import { data as mockData } from './mockData'
 import { colors } from '@static/theme'
 
 import useStyles from './style'
 interface Data {
   id: string
-  data: Array<{ x: number; y: number }>
+  points: Array<{ x: number; y: number }>
+}
+interface IProp {
+  data: Data[]
 }
 
-export const LinePlotContainer: React.FC = () => {
+enum OptionTime {
+  W = 604800000,
+  M = 2629743000,
+  Y = 31556926000
+}
+export const LinePlotContainer: React.FC<IProp> = ({ data }) => {
   const classes = useStyles()
-  const [menuOption, setMenuOption] = React.useState('Volument')
-  const [activeButtonTime, setActiveButtonTime] = React.useState('Y')
-  const [data, setData] = React.useState<Data>(mockData[0])
-  const [dataPlot, setDataPlot] = React.useState<Array<{ x: number; y: number }>>(
-    mockData[0].data.sort((a: { x: number; y: number }, b: { x: number; y: number }) => {
-      return a.x - b.x
-    })
-  )
+  const [menuOption, setMenuOption] = React.useState('Volume')
+  const [activeButtonTime, setActiveButtonTime] = React.useState<OptionTime>(OptionTime.Y)
+  const [dataTmp, setDataTmp] = React.useState<Data>(data[0])
+  const [dataPlot, setDataPlot] = React.useState<Array<{ x: number; y: number }>>(data[0].points)
   const changeData = (name: string) => {
-    const value = mockData.findIndex(element => element.id === name)
+    const value = data.findIndex(element => element.id === name)
 
-    setData(mockData[value])
+    setDataTmp(data[value])
   }
 
-  const sortData = (type: string) => {
+  const sortData = (type: OptionTime) => {
     const timestamp = Date.now()
-    if (type === 'D') {
-      setDataPlot(
-        data.data.filter(element => {
-          return element.x > timestamp - 86400000
-        })
-      )
-    }
-    if (type === 'W') {
-      setDataPlot(
-        data.data.filter(element => {
-          return element.x > timestamp - 604800000
-        })
-      )
-    }
-    if (type === 'M') {
-      setDataPlot(
-        data.data.filter(element => {
-          return element.x > timestamp - 2629743000
-        })
-      )
-    }
-    if (type === 'Y') {
-      setDataPlot(
-        data.data.filter(element => {
-          return element.x > timestamp - 31556926000
-        })
-      )
-    }
+
+    setDataPlot(
+      dataTmp.points.filter(element => {
```

Comment:
maybe you can just use here slice method with negative argument

Reply:
ok, but in the new mock-ups this function will not be needed, so I will remove it